reverse_engineering/hqlToCollectionsVisitor.js

```diff
@@ -1321,6 +1304,7 @@ class Visitor extends HiveParserVisitor {
 		};
 	}
 
+	// Check that this function is identical to the visitCreateBloomfilterIndexMainStatement
```

Comment:
To be assessed how you want to tackle that especially because semantically visiting the create and the drop statement share the exact same logic duplicated (not super obvious why because one could expect the must return different things...)

Reply:
@bigorn0 We can move duplicated logic inside these two methods to a separate method in the visitor class, but having two entry points (`visitDropBloomfilterIndexMainStatement` and `visitCreateBloomfilterIndexMainStatement`) is required by ANTLR4 grammar - we should keep them.

Reply:
Yes it is indeed required, shopuldn't we expect different information extracted et returned by the visitor function for create vs drop?

Reply:
Both methods return objects that represent BloomfilterIndex even if it's a DROP statement. As I know originally it was a requirement to parse such statements (DROP... statements) and reverse them in the Hackolade model as deactivated objects. But the BloomfilterIndexes doesn't have the isActivated property - so as a result we have the identical logic and it's expected for now I supose.